### ISSUE TITLE
`ResizeObserver is not defined` error when used with `jsdom`

### ISSUE THREAD

**Issue by gbyesiltas** [danielroe/nuxt-vitest#287]
I am using FormKit in my components, and formkit is throwing a `ResizeObserver is not defined` error when I'm running my tests with the `jsdom` environment instead of the `happy-dom` environment

**Comment by oskarols** [comment]
This is due to `ResizeObserver` observer not being present in `JSDOM` and `nuxt-vitest` doesn't do any attempt to polyfill it either. You will probably not be able to set this via `setupFiles` in Vitest either since the Nuxt environment gets created _before_ the `setupFiles` are ever run.

In general it would be really nice if there was some way of extending the DOM environment before Nuxt is run: either by `nuxt-vitest` providing some way of specifying a specific setup file for this, or by delaying the app execution until _after_ the setupFiles are run.

**Comment by OlaAlsaker** [comment]
This is a showstopper for us as well, as we are not able to mock `ResizeObserver`, `indexedDB` etc.

@danielroe is it possible to mock these things with some workarounds?

**Comment by danielroe** [comment]
Yes, you can mock things like that using vitest's `setupFiles` option, IIRC.

**Comment by OlaAlsaker** [comment]
I actually tried that, but it didn't quite work – at least not for plugins. I have a Nuxt-plugin that uses `idb-keyval` (which tries to use `indexedDB` under the hood). Even tried renaming the plugin into `plugin.client.ts`.

I assume that the mocking happens _after_ the Nuxt-environment (with plugins) is loaded?

**Comment by oskarols** [comment]
> Yes, you can mock things like that using vitest's `setupFiles` option, IIRC.

I don't think this is actually the case unfortunately. Since nuxt-vitest architecturally hooks itself up as a vitest-environment, it gets run _before_ `setupFiles` is ever run.

**Comment by OlaAlsaker** [comment]
That is correct, @oskarols . Hopefully my PR (#361) will fix this issue, as it will be able to mock stuff before the Nuxt environment.
